subsys/bluetooth/mesh/settings.h

```diff
@@ -4,29 +4,41 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-void bt_mesh_store_net(void);
-void bt_mesh_store_iv(bool only_duration);
-void bt_mesh_store_seq(void);
-void bt_mesh_store_rpl(struct bt_mesh_rpl *rpl);
-void bt_mesh_store_subnet(uint16_t net_idx);
-void bt_mesh_store_app_key(uint16_t app_idx);
-void bt_mesh_store_hb_pub(void);
-void bt_mesh_store_cfg(void);
-void bt_mesh_store_mod_bind(struct bt_mesh_model *mod);
-void bt_mesh_store_mod_sub(struct bt_mesh_model *mod);
-void bt_mesh_store_mod_pub(struct bt_mesh_model *mod);
-void bt_mesh_store_label(void);
-void bt_mesh_store_cdb(void);
-void bt_mesh_store_cdb_node(const struct bt_mesh_cdb_node *node);
-void bt_mesh_store_cdb_subnet(const struct bt_mesh_cdb_subnet *sub);
-void bt_mesh_store_cdb_app_key(const struct bt_mesh_cdb_app_key *app);
+/* Pending storage actions. */
+enum bt_mesh_settings_flag {
+	BT_MESH_SETTINGS_RPL_PENDING,
+	BT_MESH_SETTINGS_NET_KEYS_PENDING,
+	BT_MESH_SETTINGS_APP_KEYS_PENDING,
+	BT_MESH_SETTINGS_NET_PENDING,
+	BT_MESH_SETTINGS_IV_PENDING,
+	BT_MESH_SETTINGS_SEQ_PENDING,
+	BT_MESH_SETTINGS_HB_PUB_PENDING,
+	BT_MESH_SETTINGS_CFG_PENDING,
+	BT_MESH_SETTINGS_MOD_PENDING,
+	BT_MESH_SETTINGS_VA_PENDING,
+	BT_MESH_SETTINGS_CDB_PENDING,
 
-void bt_mesh_clear_net(void);
-void bt_mesh_clear_subnet(uint16_t net_idx);
-void bt_mesh_clear_app_key(uint16_t app_idx);
-void bt_mesh_clear_rpl(void);
-void bt_mesh_clear_cdb_node(struct bt_mesh_cdb_node *node);
-void bt_mesh_clear_cdb_subnet(struct bt_mesh_cdb_subnet *sub);
-void bt_mesh_clear_cdb_app_key(struct bt_mesh_cdb_app_key *app);
+	BT_MESH_SETTINGS_FLAG_COUNT,
+};
+
+#ifdef CONFIG_BT_SETTINGS
+#define MESH_SETTINGS_STATIC_HANDLER_DEFINE SETTINGS_STATIC_HANDLER_DEFINE
```

Comment:
`BT_MESH_...`. Perhaps just `BT_MESH_SETTINGS_DEFINE`?

Since we're redefining this now, perhaps it could abstract some of the duplication in each handler? E.g.

```c
#define BT_MESH_SETTINGS_DEFINE(_hname, _subtree, _set) \
    SETTINGS_STATIC_HANDLER_DEFINE(bt_mesh_##_hname, "bt/mesh/" _subtree, NULL, _set, NULL, NULL)
```
This'll also remove the pitfall of adding commit handlers, which won't work the way we expect.